prime-router/src/main/kotlin/cli/ProcessHl7Commands.kt

```diff
@@ -0,0 +1,236 @@
+            val children = input.getAll(iname)
+            children.forEachIndexed { index, c ->
+                indexStructure(c, (index + 1).toString(), inputMap)
+            }
+        }
+        val outputNames = output.names
+        outputNames.filter { name -> output.getAll(name).isNotEmpty() }.forEach { iname ->
+            val children = output.getAll(iname)
+            children.forEachIndexed { index, c ->
+                indexStructure(c, (index + 1).toString(), outputMap)
+            }
+        }
+
+        fun compareHl7Type(
+            segmentIndex: String,
+            input: Type,
+            output: Type,
+            segmentType: String,
+            segmentNumber: Int
+        ): Boolean {
+            return when {
+                input is Primitive && output is Primitive && !StringUtils.equals(input.value, output.value) -> {
+                    echo(
+                        "$segmentIndex-$segmentType number $segmentNumber did not match on ${input.value}, " +
+                            output.value
+                    )
+                    return false
+                }
+
+                input is Varies && output is Varies -> compareHl7Type(
+                    segmentIndex,
+                    input.data,
+                    output.data,
+                    segmentType,
+                    segmentNumber
+                )
+
+                input is Composite && output is Composite -> {
+                    val inputComponents = input.components.filter { !it.isEmpty }
+                    val outputComponents = output.components.filter { !it.isEmpty }
+                    val inputExtraComponents = input.extraComponents
+                    val outputExtraComponents = output.extraComponents
+                    if (inputComponents.size != outputComponents.size) {
+                        echo("$segmentIndex-$segmentType number $segmentNumber did not match")
+                        return false
+                    } else if (inputExtraComponents.numComponents() != outputExtraComponents.numComponents()) {
+                        echo("$segmentIndex-$segmentType number $segmentNumber did not match")
+                        return false
+                    } else {
+                        inputComponents.zip(outputComponents).forEach { (i, o) ->
+                            if (!compareHl7Type(segmentIndex, i, o, segmentType, segmentNumber)) {
+                                echo("$segmentIndex-$segmentType number $segmentNumber did not match on $i, $o")
+                                return false
+                            }
+                        }
+                        return true
+                    }
+                }
+
+                else -> {
+                    return true
+                }
+            }
+        }
+
+        val mapNumOfSegment = mutableMapOf<String, Int>()
+
+        inputMap.forEach { (segmentIndex, segment) ->
```

Comment:
Nit: could turn this into `reduce`

Reply:
I am not following what I can use reduce on here.

Reply:
Can we talk about this tomorrow? This one does not give reduce as an option